### PR TITLE
Updates for RC

### PR DESCRIPTION
**Pull request by paf31**
@garyb Does this look ok?

### CODE REVIEW

File: src/Test/QuickCheck/Laws/Data/BooleanAlgebra.purs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+-- | - Complemented:
+-- |   - `not a || a == top`
+-- |   - `not a && a == bottom`
+-- | - Double negation:
+-- |   - `not <<< not == id`
+checkBooleanAlgebra :: forall a. (Arbitrary a, Eq a, BooleanAlgebra a) => Proxy a -> QC Unit
+checkBooleanAlgebra _ = do
+
+  log "Checking 'Complemented' law for ComplementedLattice"
```
Comment (garyb) [inline src/Test/QuickCheck/Laws/Data/BooleanAlgebra.purs]: Probably should update these `logs` to say `BooleanAlgebra` :)

File: src/Test/QuickCheck/Laws/Data/BoundedOrd.purs [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
-checkBounded _ = do
-
-  log "Checking 'Ordering' law for Bounded"
-  quickCheck ordering
-
-  where
-
-  ordering :: a -> Boolean
-  ordering a = bottom <= a && a <= top
+module Test.QuickCheck.Laws.Data.Bounded where
```
Comment (garyb) [inline src/Test/QuickCheck/Laws/Data/BoundedOrd.purs]: We should probably make this `BoundedOrd` as the law applies to that specifically.

**Comment by paf31** [comment]
@garyb This is the best I could do while sticking to vanilla type class contexts. It tidies up the signatures quite a lot, but uglies up the internals a bit. I didn't bother with `Semigroupoid` and `Category` yet, since I wanted to get your thoughts first.

I'd totally understand if you wanted to ditch this :)

**Comment by garyb** [comment]
It will be a shame to lose the implementations that are "just the laws", but it might well be worth it for the simpler interface, as you're definitely right in saying that we shouldn't have to provide the `a, b, c` etc for a `f` or `m`.

**Comment by garyb** [comment]
The `Semigroupoid` and `Category` stuff, I'm not even sure how useful that is, I had a hard time coming up with tests for things like the `(->)` instance. I think I ended up testing the `Semigroupoid Tuple` instance but that was it. Thoughts on that would be welcome too.

**Comment by paf31** [comment]
Agreed, I think the really useful checks will be for data structures. Maybe for now, we should take the `Category` checks out, since you can't determine equality on functions easily anyway.

One possible resolution would be to have a variant on `Eq` which built up evidence for function equality, like so:

``` purescript
class EqGen a where
  eq :: a -> a -> Gen Boolean
```

Then you could define an instance for `(Arbitrary a, Eq b) => EqGen (a -> b)`.

**Pull request closed without merging**